addons/copy-reporter/userscript.js

```diff
@@ -0,0 +1,81 @@
+export default async function ({ addon, console, msg }) {
+  addon.tab.createEditorContextMenu(
+    (ctx) => {
+      // ugly hack to get the string value
+      let element = ctx.target.lastChild.lastChild;
+
+      if (ctx.type === "monitor_slider") {
+        element = element.children[element.children.length - 2].lastChild;
+      } else {
+        element = element.lastChild;
+        if (ctx.type === "monitor_default") {
+          element = element.lastChild;
+        }
+      }
+
+      if (element.innerText.length !== 0) {
+        navigator.clipboard.writeText(element.innerText);
+      }
+    },
+    {
+      className: "copy",
+      types: ["monitor_default", "monitor_large", "monitor_slider"],
+      position: "monitor",
+      order: 0,
+      label: msg("copy-value"),
+    }
+  );
+
+  // add button to reporter bubble
+  const ScratchBlocks = await addon.tab.traps.getBlockly();
+
+  // https://github.com/LLK/scratch-blocks/blob/893c7e7ad5bfb416eaed75d9a1c93bdce84e36ab/core/workspace_svg.js#L979
+  ScratchBlocks.WorkspaceSvg.prototype.reportValue = function (id, value) {
+    let block = this.getBlockById(id);
+    if (!block) {
+      throw "Tried to report value on block that does not exist.";
+    }
+
+    ScratchBlocks.DropDownDiv.hideWithoutAnimation();
+    ScratchBlocks.DropDownDiv.clearContent();
+
+    let contentDiv = ScratchBlocks.DropDownDiv.getContentDiv();
+
+    let valueReportBox = document.createElement("div");
+    valueReportBox.setAttribute("class", "valueReportBox");
+    valueReportBox.innerHTML = ScratchBlocks.scratchBlocksUtils.encodeEntities(value);
+    valueReportBox.style.userSelect = "text";
+    // use to get focus and event priority
+    valueReportBox.setAttribute("tabindex", "0");
+    // if the user pressed Ctrl+C, prevent propagation to Blockly
+    valueReportBox.onkeydown = (event) => {
+      if ((event.altKey || event.ctrlKey || event.metaKey) && event.keyCode == 67) {
```

Comment:
Is there a reason for the Alt and Windows key checks?
```suggestion
      if (event.ctrlKey && event.key === "c") {
```

Reply:
This is based on the Blockly source.

Reply:
`scratch-blocks`? That's kind of a mess and I think [keyCode is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode).

Reply:
`event.keyCode == 67` can be replaced with `event.code == "KeyC"`.

It would be easier and more reliable to simply stop propagation of all `keydown` events though.

Reply:
It's designed to only prevent the events which scratch-blocks reads as copying a block.

Reply:
I will replace it with event.key === "c" though as that's the same behaviour but not deprecated. I didn't realise.

Reply:
> I will replace it with event.key === "c" though as that's the same behaviour but not deprecated. I didn't realise.

It's not the same. On some keyboard layouts, the C key is in a different position. Its `key` is still `"c"`, but the `keyCode` is different.

Reply:
Does "KeyC" work like before?

Reply:
> Does "KeyC" work like before?

Yes, unless there are weird edge cases where it behaves differently.

Reply:
On my browser e.key seems to be lowercase C though which is weird.

Reply:
`event.key` is the character ("c"). `event.code` is the name of the key ("KeyC").

Reply:
I got them confused. I can fix this (probably)!